src/test_utils.jl

```diff
@@ -0,0 +1,135 @@
+module TestUtils
+
+const __ATOL = 1e-9
+
+using LinearAlgebra
+using KernelFunctions
+using Random
+using Test
+
+"""
+    test_interface(
+        k::Kernel,
+        x0::AbstractVector,
+        x1::AbstractVector,
+        x2::AbstractVector;
+        atol=__ATOL,
+    )
+
+Run various consistency checks on `k` at the inputs `x0`, `x1`, and `x2`.
+`x0` and `x1` should be of the same length with different values, while `x0` and `x2` should
+be of different lengths.
+
+    test_interface([rng::AbstractRNG], k::Kernel, T::Type{<:AbstractVector}; atol=__ATOL)
+
+`test_interface` offers certain types of test data generation to make running these tests
+require less code for common input types. For example, `Vector{<:Real}`, `ColVecs{<:Real}`,
+and `RowVecs{<:Real}` are supported. For other input vector types, please provide the data
+manually. 
+"""
+function test_interface(
+    k::Kernel,
+    x0::AbstractVector,
+    x1::AbstractVector,
+    x2::AbstractVector;
+    atol=__ATOL,
+)
+    # TODO: uncomment the tests of ternary kerneldiagmatrix.
+
+    # Ensure that we have the required inputs.
+    @assert length(x0) == length(x1)
+    @assert length(x0) ≠ length(x2)
+
+    # Check that kerneldiagmatrix basically works.
+    # @test kerneldiagmatrix(k, x0, x1) isa AbstractVector
+    # @test length(kerneldiagmatrix(k, x0, x1)) == length(x0)
+
+    # Check that pairwise basically works.
+    @test kernelmatrix(k, x0, x2) isa AbstractMatrix
+    @test size(kernelmatrix(k, x0, x2)) == (length(x0), length(x2))
+
+    # Check that elementwise is consistent with pairwise.
+    # @test kerneldiagmatrix(k, x0, x1) ≈ diag(kernelmatrix(k, x0, x1)) atol=atol
+
+    # Check additional binary elementwise properties for kernels.
+    # @test kerneldiagmatrix(k, x0, x1) ≈ kerneldiagmatrix(k, x1, x0)
+    @test kernelmatrix(k, x0, x2) ≈ kernelmatrix(k, x2, x0)' atol=atol
+
+    # Check that unary elementwise basically works.
+    @test kerneldiagmatrix(k, x0) isa AbstractVector
+    @test length(kerneldiagmatrix(k, x0)) == length(x0)
+
+    # Check that unary pairwise basically works.
+    @test kernelmatrix(k, x0) isa AbstractMatrix
+    @test size(kernelmatrix(k, x0)) == (length(x0), length(x0))
+    @test kernelmatrix(k, x0) ≈ kernelmatrix(k, x0)' atol=atol
+
+    # Check that unary elementwise is consistent with unary pairwise.
+    @test kerneldiagmatrix(k, x0) ≈ diag(kernelmatrix(k, x0)) atol=atol
+
+    # Check that unary pairwise produces a positive definite matrix (approximately).
+    @test all(eigvals(Matrix(kernelmatrix(k, x0))) .> -atol)
```

Comment:
An (IMO simpler) alternative:
```suggestion
    @test eigmin(Matrix(kernelmatrix(k, x0))) > -atol
```